### PR TITLE
Text modal

### PR DESCRIPTION
**Pull request by Fanuel-D**
Summary of Changes

**Technical Challenge 1 complete** 
• designed a scalable question and answering system that encourages active recall when learning something new. 
• It is capable of handling multiple question inputs from the user and effectively synchronizing them to ask them at the right time and interval. 
• Based on ease of answering a question and whether or not the question is answered correctly or not, the next recall time is adjusted to encourage a stronger retention of material.

**Implemented the Reflections page with reflection entries** 
• Whenever a user leaves a study session, they will be prompted to  add reflection on material they have just covered. 
This feature is designed this way to better help readers assimilate the content they just read. 
Screenshot 2024-07-11 at 10 11 28 PM
Screenshot 2024-07-11 at 10 11 38 PM
Screenshot 2024-07-11 at 10 11 51 PM

### CODE REVIEW

**Comment by Stannie-Lim** [comment]
Looks like you're just pushing to the same branch, then merging into the same branch text-modal over and over again

Usually with every single new feature, you should create a new branch for that feature, then merge it in

Here's the full flow:

// start off on your main branch
git checkout -b navbar

// code a bit

git add .
git commit -m 'did some stuff'
git push -u origin navbar

// make a pull request on github, merge it in

git checkout main
git pull

// start from the beginning but make a different feature branch

File: frontend/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "pdfjs-dist": "^4.3.136",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
+    "react-modal": "^3.16.1",
```
Comment (Stannie-Lim) [inline frontend/package.json:26]: if you want you can prob just use mui dialog (´・◡・｀)

File: frontend/src/components/FilePage.jsx [hunk 2]
```diff
@@ -11,6 +11,8 @@ import Button from "@mui/material/Button";
 import QuestionBankModal from "./QuestionBankModal";
 import SideBar from "./SideBar";
 import SectionBox from "./SectionBox";
+import QuestionModal from "./QuestionModal";
+import RecallQuestion from "./RecallQuestion";
 
 function FilePage({ user, userFiles }) {
   const [userFileDetails, setUserFileDetails] = useState(null);
```
Comment (Stannie-Lim) [inline frontend/src/components/FilePage.jsx:18]: this is quite a few states back to back- wonder if there's a way to separate it out into different components some more? or consolidate it into an object state instead?

like this:

```js
const [position, setPosition] = useState({
  x: null,
  y: null
});
```

File: frontend/src/components/Reflection.jsx [hunk 3]
```diff
@@ -1,8 +1,183 @@
...
+    const title = formData.get("title");
+    try {
+      const response = await axios.post(
+        `http://localhost:3000/reflection/${fileId}`,
+        {
+          text,
+          title,
+        }
+      );
+      console.log("Success:", response.data);
```
Comment (Stannie-Lim) [inline frontend/src/components/Reflection.jsx:31]: what do you want to do after posting the reflection? if you don't do anything here, the reflection you just posted wont appear in the UI until you refresh the page, so usually you'd want to update the state with the newly created reflection after posting

File: frontend/src/components/Reflection.jsx [hunk 4]
```diff
@@ -1,8 +1,183 @@
...
+    } catch (error) {
+      console.error("Error:", error);
+    }
+  };
+
+  const handleAddClicked = () => {
+    setIsOpen(true);
+  };
+  const formatDateTime = (currDateTime) => {
```
Comment (Stannie-Lim) [inline frontend/src/components/Reflection.jsx:40]: you can use `dayjs` library for this https://day.js.org/docs/en/display/format

File: frontend/src/components/Reflection.jsx [hunk 5]
```diff
@@ -1,8 +1,183 @@
...
+          },
+          overlay: {
+            backgroundColor: "rgba(0, 0, 0, 0.3)",
+            zIndex: "3000",
+          },
+        }}
+        isOpen={isOpen}
+        onRequestClose={() => setIsOpen(false)}
+      >
+        <form
```
Comment (Stannie-Lim) [inline frontend/src/components/Reflection.jsx:92]: definitely separate these into separate components (´・◡・｀) this is a lot of child components in one modal and 1 file

File: frontend/package.json (hunk 1 above)
Comment (Fanuel-D) [reply frontend/package.json:26]: Yeah react modals are very verbose , it's almost the same as writing the modal code yourself . I just wanted to test them out, but yeah mui dialog is my goto from now on.

File: frontend/src/components/Reflection.jsx (hunk 3 above)
Comment (Fanuel-D) [reply frontend/src/components/Reflection.jsx:31]: resolved this issue, I added a dependency variable that triggers the useEffect whenever there is a submission, let me know if there is a more optimal way.

**Comment by Fanuel-D** [comment]
> Looks like you're just pushing to the same branch, then merging into the same branch text-modal over and over again
> 
> Usually with every single new feature, you should create a new branch for that feature, then merge it in
> 
> Here's the full flow:
> 
> // start off on your main branch
> 
> git checkout -b navbar
> 
> // code a bit
> 
> git add . git commit -m 'did some stuff' git push -u origin navbar
> 
> // make a pull request on github, merge it in
> 
> git checkout main git pull
> 
> // start from the beginning but make a different feature branch

File: frontend/src/components/Reflection.jsx (hunk 5 above)
Comment (Fanuel-D) [reply frontend/src/components/Reflection.jsx:92]: Resolved

File: frontend/src/components/Reflection.jsx (hunk 4 above)
Comment (Fanuel-D) [reply frontend/src/components/Reflection.jsx:40]: Resolved

File: frontend/package.json (hunk 1 above)
Comment (Fanuel-D) [reply frontend/package.json:26]: Changed to MUi dialog box

File: frontend/src/components/FilePage.jsx (hunk 2 above)
Comment (Fanuel-D) [reply frontend/src/components/FilePage.jsx:18]: Managed to separate a couple of them to different components.

**Pull request merged**